### DISCUSSION TITLE
Javascript visualization cells

### DISCUSSION THREAD

**Discussion by JD557** [polynote/polynote#1284] [Ideas]
I was wondering: would it be possible to have Javascript (or even better: Scala.js) visualization cells (similar to Vega cells)?

This could potentially be useful to have dynamic/interactive visualizations.

**Comment by jeremyrsmith** [comment]
I think so. The vega cells are basically JavaScript cells, just constrained.

Not sure about libraries though – like how we'd implement e.g. npm dependencies being available to a JS cell. Would it still be useful if you couldn't have any dependencies (other than say, d3 which I think is already loaded)?

**Reply by JD557** [reply]
> Not sure about libraries though – like how we'd implement e.g. npm dependencies being available to a JS cell.

I guess the simplest solution would be to add support raw URLs (and maybe a hash) to load libraries from somewhere like [CDNJS](https://cdnjs.com/)?

I know this is not as clean or "cool" as having the backend running something like npm/yarn (or the scala.js compiler) and returning a compiled JS blob, but it should be much simpler to implement for a first version.
